### PR TITLE
Deckimporter: Updated handling for parallel, revised and promo investigators

### PR DESCRIPTION
**Pull request by Chr1Z93**
Since the revised and promo art for investigators are no longer added as separate cards (but states instead) this updates the deckimporter to automatically load the correct state for these cards.

### CODE REVIEW

**Comment by Tikatoy** [comment]
Are the revised core art investigators  in the mod?

Or was this for promo investigator art? I tested with an old Norman Withers deck that has the promo art, and it did load the Norman Withers investigator card correctly, and I was able to change state.

**Comment by Chr1Z93** [comment]
> Are the revised core art investigators in the mod?
> 
> Or was this for promo investigator art? I tested with an old Norman Withers deck that has the promo art, and it did load the Norman Withers investigator card correctly, and I was able to change state.

Yeah, the alternative arts (revised and promo) are added as states to the normal cards. When I did this for the last update, I didn't account for the deck importer. The bug at hand is that it still tries to load the revised investigator cards by ID. This PR replaces that ID changing for revised cards with a message that those can be found in the state. (If this didn't make sense, try loading a revised Daisy deck in 3.0.0 and then in the version from this mod.)

**Comment by Tikatoy** [comment]
I created a Daisy deck selecting Revised Core for  front and back: https://arkhamdb.com/deck/view/2800646

In both 3.0.0 and the save I generated from the branch, it loaded the (stateless) parallel front/back Daisy. Do I need to be doing something different?

**Comment by Chr1Z93** [comment]
> I created a Daisy deck selecting Revised Core for front and back: https://arkhamdb.com/deck/view/2800646
> 
> In both 3.0.0 and the save I generated from the branch, it loaded the (stateless) parallel front/back Daisy. Do I need to be doing something different?

Yep, that's the issue. It should not be loading the parallel ones if you select revised front/back.

**Comment by Chr1Z93** [comment]
> I created a Daisy deck selecting Revised Core for front and back: https://arkhamdb.com/deck/view/2800646
> 
> In both 3.0.0 and the save I generated from the branch, it loaded the (stateless) parallel front/back Daisy. Do I need to be doing something different?

Uhh that's a good catch. I only tested revised front and didn't add revised back. Will need to update this PR

**Comment by Chr1Z93** [comment]
Pr is updated and should now handle alt art correctly. I didn't have an old decklist with promo art available for testing, but assume that that will be analog to the revised art.

File: src/arkhamdb/DeckImporterMain.ttslua [hunk 1]
```diff
@@ -407,6 +410,51 @@ function handleCustomizableUpgrades(cardList, customizations)
...
+---@param card Object Card which needs to be set the state for
+---@param loadAltInvestigator String Contains the name of alternative art for the investigator ("normal", "revised" or "promo")
+function loadAltArt(card, loadAltInvestigator)
+  if loadAltInvestigator == "normal" then return end
+
+  -- lookup correct stateId for investigator and alt art state
+  local baseId = string.gsub(JSON.decode(card.getGMNotes()).id, "-m", "")
+
+  local stateIdTable = {}
+  -- Roland Banks
```
Comment (Buhallin) [inline src/arkhamdb/DeckImporterMain.ttslua:423]: I don't love the hardcoding here.

Could do this by adding something to the states - maybe Memo, maybe an extra metadata field - with the "revised" or "promo" tag, then inspect the card data to find the correct one.  Would be more durable for future changes, but it's probably not worth it given the limited number of alt arts,

**Review by Buhallin — APPROVED** [review]

**Pull request merged**